ci/runner/Dockerfile

```diff
@@ -0,0 +1,77 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Note this is the Dockerfile for the CI runner not for SRF itself
+# Build from the root of the SRF repo with:
+# docker build --network=host -t srf/ci:latest -f ci/runner/Dockerfile .
+
+# Args used in FROM commands must come first
+ARG FROM_IMAGE="rapidsai/ci"
+ARG CUDA_VER=11.4.1
+ARG LINUX_DISTRO=ubuntu
+ARG LINUX_VER=20.04
+ARG PYTHON_VER=3.8
+
+# Configure the base docker img
+FROM ${FROM_IMAGE}:cuda${CUDA_VER}-${LINUX_DISTRO}${LINUX_VER}-py${PYTHON_VER} AS base
+
+ARG PROJ_NAME=srf
+ARG CUDA_SHORT_VER=11.4
+ARG CLANG_VER=12
+
+SHELL ["/bin/bash",  "-c"]
+
+# OS deps
+RUN apt-get update &&\
+    apt-get install --no-install-recommends -y \
+        clang-format-${CLANG_VER} \
+        clang-tidy-${CLANG_VER} \
+        libclang-${CLANG_VER}-dev \
+        libnuma1 \
+        libnvidia-compute-495 \
+        llvm-${CLANG_VER}-dev \
+    && rm -rf /var/lib/apt/lists/* &&\
+    update-alternatives --install /usr/bin/clang clang /usr/bin/clang-${CLANG_VER} 121 \
+        --slave /usr/bin/clang-apply-replacements clang-apply-replacements /usr/bin/clang-apply-replacements-${CLANG_VER} \
+        --slave /usr/bin/clang-format clang-format /usr/bin/clang-format-${CLANG_VER} \
+        --slave /usr/bin/clang-format-diff clang-format-diff /usr/bin/clang-format-diff-${CLANG_VER} \
+        --slave /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-${CLANG_VER} \
+        --slave /usr/bin/clang-tidy-diff.py clang-tidy-diff.py /usr/bin/clang-tidy-diff-${CLANG_VER}.py \
+        --slave /usr/bin/clang++ clang++ /usr/bin/clang++-${CLANG_VER} \
+        --slave /usr/bin/git-clang-format git-clang-format /usr/bin/git-clang-format-${CLANG_VER}
+
+# Create conda environment
+COPY ./ci/conda/environments/dev_env.yml /tmp
+
+RUN CONDA_ALWAYS_YES=true /opt/conda/bin/mamba env create -n ${PROJ_NAME} -q --file /tmp/dev_env.yml && \
+    sed -i "s/conda activate base/conda activate ${PROJ_NAME}/g" ~/.bashrc && \
+    conda clean -afy && \
+    rm /tmp/dev_env.yml
+
+
+# Install IWYU
+RUN git clone https://github.com/include-what-you-use/include-what-you-use.git /tmp/iwyu &&\
+    cd /tmp/iwyu &&\
+    git checkout clang_${CLANG_VER} &&\
+    source activate ${PROJ_NAME} && \
+    cmake -G Ninja \
+        -DCMAKE_C_COMPILER=$(which clang) \
+        -DCMAKE_CXX_COMPILER=$(which clang++) \
+        -DCMAKE_INSTALL_PREFIX=${CONDA_PREFIX}  . && \
+    cmake --build . --parallel $(nproc) --target install && \
+    cd - && \
+    rm -rf /tmp/iwyu
```

Comment:
Same comment here. We can install this now via conda.

Reply:
The version in conda requires clang-14, which we aren't on until Ryan's PR #156 is merged.